charts/opentelemetry-operator/templates/_helpers.tpl

```diff
@@ -159,15 +159,11 @@ The image to use for opentelemetry-operator.
 {{- end }}
 
 {{- define "opentelemetry-operator.featureGatesMap" -}}
-{{$first := true}}
-{{- range $key, $value := .Values.manager.featureGatesMap -}}
-    {{- if $first -}}
-        {{ $first = false }}
-    {{- else -}}
-        ,
-    {{- end -}}
-    {{- if $value -}}
-        {{- $key }}
-    {{- end -}}
+{{- $list := list -}}
+{{- range $k, $v := .Values.manager.featureGatesMap -}}
+{{- if $v -}}
+{{- $list = append $list (printf "%s=true" $k) -}}
```

Comment:
If someone explicitly set `asdsadsa.adsads2: false` we should disable that feature gate. Let's add an `else` case to set `-asdsadsa.adsads2`